velox/functions/prestosql/aggregates/tests/MaxSizeForStatsTest.cpp

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::aggregate::test {
+
+namespace {
+
+class MaxSizeForStatsTest : public AggregationTestBase {
+ protected:
+};
+
+// Input:
+// c0(bigint)|c1(tinyint)|c2(smallint)|c3(integer)|c4(bigint)|c5(real)|c6(double)
+// rand()|rand()|rand()|rand()|rand()|rand()|rand()
+// Query:
+// select aggr(c1),aggr(c2),aggr(c3),aggr(c4),aggr(c5),aggr(c6);
+// Output
+// 1|2|4|8|4|8
+TEST_F(MaxSizeForStatsTest, allScalarTypes) {
+  auto vectors = {makeRowVector(
+      {makeFlatVector<int64_t>(1),
+       makeFlatVector<int8_t>(1),
+       makeFlatVector<int16_t>(1),
+       makeFlatVector<int32_t>(1),
+       makeFlatVector<int64_t>(1),
+       makeFlatVector<float>(1),
+       makeFlatVector<double>(1)})};
+  // Global partial + final aggregation.
+  {
+    auto agg = PlanBuilder()
+                   .values(vectors)
+                   .partialAggregation(
+                       {},
+                       {"\"$internal$max_data_size_for_stats\"(c1)",
+                        "\"$internal$max_data_size_for_stats\"(c2)",
+                        "\"$internal$max_data_size_for_stats\"(c3)",
+                        "\"$internal$max_data_size_for_stats\"(c4)",
+                        "\"$internal$max_data_size_for_stats\"(c5)",
+                        "\"$internal$max_data_size_for_stats\"(c6)"})
+                   .finalAggregation()
+                   .planNode();
+    assertQuery(
+        agg,
+        makeRowVector({
```

Comment:
this might be easier to read as a SQL query: 

`VALUES (1), (2), (4), (8), (4), (8)`